--- a/src/perl/common/Core.xs
+++ b/src/perl/common/Core.xs
@@ -237,12 +237,7 @@ timeout_add(msecs, func, data)
 	SV *func
 	SV *data
 CODE:
-	if (msecs < 10) {
-		croak("Irssi::timeout() : msecs must be >= 10");
-		RETVAL = -1;
-	} else {
-		RETVAL = perl_timeout_add(msecs, func, data, FALSE);
-	}
+	RETVAL = perl_timeout_add(msecs, func, data, FALSE);
 OUTPUT:
 	RETVAL
 
@@ -252,12 +247,7 @@ timeout_add_once(msecs, func, data)
 	SV *func
 	SV *data
 CODE:
-	if (msecs < 10) {
-		croak("Irssi::timeout_once() : msecs must be >= 10");
-		RETVAL = -1;
-	} else {
-		RETVAL = perl_timeout_add(msecs, func, data, TRUE);
-	}
+	RETVAL = perl_timeout_add(msecs, func, data, TRUE);
 OUTPUT:
 	RETVAL
 
